messaging/src/main/java/org/axonframework/deadline/jobrunr/JobRunrDeadlineManager.java

```diff
@@ -104,62 +113,74 @@ protected JobRunrDeadlineManager(Builder builder) {
         this.spanFactory = builder.spanFactory;
     }
 
+    /**
+     * Provides the class to use in spans,
+     *
+     * @return the class name
+     */
+    @SuppressWarnings("squid:S3400")
+    protected String getSpanClassName() {
+        return "JobRunrDeadlineManager";
+    }
+
     @Override
     public String schedule(@Nonnull Instant triggerDateTime, @Nonnull String deadlineName,
                            @Nullable Object messageOrPayload,
                            @Nonnull ScopeDescriptor deadlineScope) {
         DeadlineMessage<Object> deadlineMessage = asDeadlineMessage(deadlineName, messageOrPayload, triggerDateTime);
         UUID deadlineId = UUID.randomUUID();
-        Span span = spanFactory.createDispatchSpan(() -> "JobRunrDeadlineManager.schedule(" + deadlineName + ")",
+        Span span = spanFactory.createDispatchSpan(() -> getSpanClassName() + ".schedule(" + deadlineName + ")",
                                                    deadlineMessage);
         runOnPrepareCommitOrNow(span.wrapRunnable(() -> {
             DeadlineMessage<Object> interceptedDeadlineMessage = processDispatchInterceptors(deadlineMessage);
-            String deadlineDetails = DeadlineDetails.serialized(deadlineName,
-                                                                deadlineScope,
-                                                                interceptedDeadlineMessage,
-                                                                serializer);
-            jobScheduler.<JobRunrDeadlineManager>schedule(deadlineId,
-                                                          triggerDateTime,
-                                                          deadlineManager -> deadlineManager.execute(deadlineDetails));
+            String serializedDeadlineDetails = DeadlineDetails.serialized(
+                    deadlineName,
+                    deadlineScope,
+                    interceptedDeadlineMessage,
+                    serializer);
+            String combinedLabel = getCombinedLabel(serializer, deadlineName, deadlineScope);
+            JobBuilder job = JobBuilder.aJob()
+                                       .withId(deadlineId)
+                                       .withName(deadlineName)
+                                       .withLabels(getLabel(deadlineName), combinedLabel)
+                                       .withDetails(() -> this.execute(serializedDeadlineDetails))
```

Comment:
Funny to note that the method to invoke is defined as "details" of the job. Sounds like something more important than just details to me.

Reply:
Indeed, not that intuitive.